packages/@n8n/db/src/entities/__tests__/user.entity.test.ts

```diff
@@ -32,4 +32,242 @@ describe('User Entity', () => {
+				expect(() => user.preUpsertHook()).toThrow(
+					'Cannot save user <testexample.com>: Provided email is invalid',
+				);
+			});
+
+			it('should throw Error for email without local part', () => {
+				const user = new User();
+				user.email = '@example.com';
+
+				expect(() => user.preUpsertHook()).toThrow(Error);
+				expect(() => user.preUpsertHook()).toThrow(
+					'Cannot save user <@example.com>: Provided email is invalid',
+				);
+			});
+
+			it('should throw Error for various invalid formats', () => {
+				const invalidEmails = [
+					'test..email@example.com',
+					'test@',
+					'@example.com',
+					'test@@example.com',
+					'test @example.com',
+					'test@ example.com',
+					'.test@example.com',
+					'test.@example.com',
+					'test@example.',
+					'test@.example.com',
+				];
+
+				invalidEmails.forEach((email) => {
+					const user = new User();
+					user.email = email;
+
+					expect(() => user.preUpsertHook()).toThrow(Error);
+					expect(() => user.preUpsertHook()).toThrow(
+						`Cannot save user <${email}>: Provided email is invalid`,
+					);
+				});
+			});
+		});
+
+		describe('edge cases', () => {
+			it('should throw Error for empty string email', () => {
+				const user = new User();
+				user.email = '';
+
+				expect(() => user.preUpsertHook()).toThrow(Error);
+				expect(() => user.preUpsertHook()).toThrow(
+					'Cannot save user <>: Provided email is invalid',
+				);
+			});
+
+			it('should throw Error for whitespace-only email', () => {
+				const user = new User();
+				user.email = '   ';
+
+				expect(() => user.preUpsertHook()).toThrow(Error);
+				expect(() => user.preUpsertHook()).toThrow(
+					'Cannot save user <   >: Provided email is invalid',
+				);
+			});
+
+			it('should throw Error for very long invalid email', () => {
+				const user = new User();
+				const longInvalidEmail = 'a'.repeat(300) + '@invalid';
+				user.email = longInvalidEmail;
+
+				expect(() => user.preUpsertHook()).toThrow(Error);
+				expect(() => user.preUpsertHook()).toThrow(
+					`Cannot save user <${longInvalidEmail}>: Provided email is invalid`,
+				);
+			});
+
+			it('should throw Error for emails with special characters that should fail', () => {
+				const invalidEmailsWithSpecialChars = [
+					'test<>@example.com',
+					'test[]@example.com',
+					'test()@example.com',
+					'test,@example.com',
+					'test;@example.com',
+					'test:@example.com',
+					'test"@example.com',
+					'test\\@example.com',
+				];
+
+				invalidEmailsWithSpecialChars.forEach((email) => {
+					const user = new User();
+					user.email = email;
+
+					expect(() => user.preUpsertHook()).toThrow(Error);
+					expect(() => user.preUpsertHook()).toThrow(
+						`Cannot save user <${email}>: Provided email is invalid`,
+					);
+				});
+			});
+		});
+
+		describe('BeforeInsert and BeforeUpdate triggers', () => {
+			it('should validate email during entity creation (BeforeInsert)', () => {
```

Comment:
I'm not sure any of those test actually do something else than the previous 'invalid email scenarios' tests